src/dbnode/integration/wide_query_test.go

```diff
@@ -0,0 +1,343 @@
+				if allowed == shard {
+					shardInUse = true
+					break
+				}
+			}
+
+			if !shardInUse {
+				continue
+			}
+		}
+
+		found := false
+		for idx, sharded := range shardedChecksums {
+			if shard != sharded.shard {
+				continue
+			}
+
+			found = true
+			shardedChecksums[idx].checksums = append(sharded.checksums, checksum)
+			break
+		}
+
+		if found {
+			continue
+		}
+
+		count++
+		shardedChecksums = append(shardedChecksums, shardedIndexChecksum{
+			shard:     shard,
+			checksums: []ident.IndexChecksum{checksum},
+		})
+	}
+
+	sort.Slice(shardedChecksums, func(i, j int) bool {
+		return shardedChecksums[i].shard < shardedChecksums[j].shard
+	})
+
+	checksums := make([]ident.IndexChecksum, 0, count)
+	for _, sharded := range shardedChecksums {
+		checksums = append(checksums, sharded.checksums...)
+	}
+
+	return checksums
+}
+
+func TestWideFetch(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow() // Just skip if we're doing a short run
+	}
+
+	var (
+		batchSize     = 7
+		seriesCount   = 100
+		blockSize     = time.Hour * 2
+		verifyTimeout = 2 * time.Minute
+	)
+
+	// Test setup
+	idxOpts := namespace.NewIndexOptions().
+		SetEnabled(true).
+		SetBlockSize(time.Hour * 2)
+	nsOpts := namespace.NewOptions().
+		SetIndexOptions(idxOpts).
+		SetRepairEnabled(false).
+		SetRetentionOptions(defaultIntegrationTestRetentionOpts)
+
+	nsID := testNamespaces[0]
+	nsMetadata, err := namespace.NewMetadata(nsID, nsOpts)
+	require.NoError(t, err)
+
+	// Set up file path prefix
+	postfix := atomic.AddUint64(&created, 1) - 1
+	filePathPrefix, err := ioutil.TempDir("", fmt.Sprintf("integration-test-%d", postfix))
+	require.NoError(t, err)
+
+	testOpts := NewTestOptions(t).
+		SetTickMinimumInterval(time.Second).
+		SetNamespaces([]namespace.Metadata{nsMetadata}).
+		SetFilePathPrefix(filePathPrefix)
+
+	require.NoError(t, err)
+	fsOpts := fs.NewOptions().SetFilePathPrefix(filePathPrefix)
+	decOpts := fsOpts.DecodingOptions().SetIndexEntryHasher(xtest.NewParsedIndexHasher(t))
+	fsOpts = fsOpts.SetDecodingOptions(decOpts)
+
+	testSetup, err := NewTestSetup(t, testOpts, fsOpts,
+		func(opt storage.Options) storage.Options {
+			return opt.SetWideBatchSize(batchSize)
+		})
+	require.NoError(t, err)
+	defer testSetup.Close()
+
+	// Start the server with filesystem bootstrapper
+	log := testSetup.StorageOpts().InstrumentOptions().Logger()
+	log.Debug("wide fetch test")
+	require.NoError(t, testSetup.StartServer())
+	log.Debug("server is now up")
+
+	// Stop the server
+	defer func() {
+		require.NoError(t, testSetup.StopServer())
+		log.Debug("server is now down")
+	}()
+
+	// Setup test data
+	now := testSetup.NowFn()()
+	indexWrites := make(TestIndexWrites, 0, seriesCount)
+	ids := make([]string, 0, seriesCount)
+	for i := 0; i < seriesCount; i++ {
+		// Keep in lex order.
+		padCount := i / 10
+		pad := ""
+		for i := 0; i < padCount; i++ {
+			pad = fmt.Sprintf("%so", pad)
+		}
+
+		id := fmt.Sprintf("foo%s-%d", pad, i)
```

Comment:
Suggesting something simpler:
```suggestion
              id := fmt.Sprintf("foo-%05d", i)
```

Reply:
Good idea 👍